src/main/java/com/minecolonies/api/tileentities/storageblocks/IStorageBlockInterface.java

```diff
@@ -0,0 +1,117 @@
+package com.minecolonies.api.tileentities.storageblocks;
+
+import com.minecolonies.api.colony.buildings.IBuilding;
+import com.minecolonies.api.crafting.ItemStorage;
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.items.IItemHandlerModifiable;
+
+import java.util.Map;
+import java.util.function.Predicate;
+
+public interface IStorageBlockInterface
+{
+    /**
+     * Whether the storageblock should be included in building containers automatically.
+     *
+     * @param building The building that is being constructed.
+     * @return Whether the storageblock should be included in building containers automatically.
+     */
+    boolean shouldAutomaticallyAdd(final IBuilding building);
```

Comment:
Also this method feels a bit weird, this implies there's a manual way to add storage blocks.
I'd leave this to the match function inside the registry entry.

Reply:
My intention is to have manual ways to add storage blocks for my addon using a scepter

Reply:
Then I'd handle this inside your matches function for the storage block. Just return false for any block unless it's added in the allow list.
Additionally you can add containers after registration by accessing the container list in the building.